### PR TITLE
docs: enable correct types for properties page in patternhub

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

Add correct types for properties

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/components/src/components/link/model.ts [hunk 1]
```diff
@@ -16,9 +16,17 @@ export const LinkContentList = ['external', 'internal'] as const;
 export type LinkContentType = (typeof LinkContentList)[number];
 
 export type DBLinkDefaultProps = {
+	/**
+	 * Adds a different arrow to after the link to show external or internal link
```
Comment (mfranzke) [inline packages/components/src/components/link/model.ts:20]:
```suggestion
	 * Adds a different arrow after the link to show external or internal link
```

File: packages/components/src/components/input/model.ts [hunk 2]
```diff
@@ -56,6 +62,9 @@ export type DBInputDefaultProps = {
 	 * 	Type of form control
 	 */
 	type?: InputTypeType | string;
+	/**
+	 * Sets step based in [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step).
```
Comment (mfranzke) [inline packages/components/src/components/input/model.ts:66]:
```suggestion
	 * Sets [step value](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step).
```

File: packages/components/src/components/input/model.ts [hunk 3]
```diff
@@ -37,7 +37,13 @@ export const InputTypeList = [
 export type InputTypeType = (typeof InputTypeList)[number];
 
 export type DBInputDefaultProps = {
+	/**
+	 * Set a data list via attributes instead of child.
+	 */
 	dataList?: string[] | ValueLabelType[];
+	/**
+	 * Add a custom id to data list if your using `dataList` attribute.
```
Comment (mfranzke) [inline packages/components/src/components/input/model.ts:45]:
```suggestion
	 * Add a custom id to data list if you're using `dataList` attribute.
```

File: packages/components/src/components/input/model.ts [hunk 4]
```diff
@@ -37,7 +37,13 @@ export const InputTypeList = [
 export type InputTypeType = (typeof InputTypeList)[number];
 
 export type DBInputDefaultProps = {
+	/**
+	 * Set a data list via attributes instead of child.
```
Comment (mfranzke) [inline packages/components/src/components/input/model.ts:41]:
```suggestion
	 * Set a [data list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist) via attributes instead of child.
```

File: packages/components/src/components/header/model.ts [hunk 5]
```diff
@@ -9,10 +9,32 @@ import {
...
+	 */
 	metaNavigation?: unknown;
+	/**
+	 * Slot to pass one or more elements like DBButton (e.g. search) as primary action.
+	 * Desktop: Shown next to the main-navigation
+	 * Mobile: Shown next to the brand
+	 */
 	primaryAction?: unknown;
+	/**
+	 * Slot to pass one or more element like DBButton (e.g. profile, language, etc.) as secondary action.
```
Comment (mfranzke) [inline packages/components/src/components/header/model.ts:29]:
```suggestion
	 * Slot to pass one or more elements like DBButton (e.g. profile, language, etc.) as secondary action.
```

File: packages/components/src/components/header/model.ts [hunk 6]
```diff
@@ -9,10 +9,32 @@ import {
...
+	 * Desktop: Shown seperated by divider at the end of the header
+	 * Mobile: Shown inside the drawer at the bottom.
+	 */
 	secondaryAction?: unknown;
+
+	/**
+	 * Open/closes the drawer on for mobile header or if `forceMobile` is true;
```
Comment (mfranzke) [inline packages/components/src/components/header/model.ts:36]:
```suggestion
	 * Open/closes the drawer for mobile header or if `forceMobile` is true.
```

File: packages/components/src/components/card/model.ts [hunk 7]
```diff
@@ -18,6 +18,9 @@ export type DBCardDefaultProps = {
 	 */
 	behaviour?: CardBehaviourType;
 
+	/**
+	 * Changes the elevation of the card which is equal to basic-background-level
```
Comment (mfranzke) [inline packages/components/src/components/card/model.ts:22]:
```suggestion
	 * Changes the elevation of the card which is equal to `basic-background-level`
```

File: packages/components/package.json [hunk 8]
```diff
@@ -17,8 +17,8 @@
   "scripts": {
     "build": "npm-run-all -p build-components build-assets -s build-style:*",
     "build-assets": "cpr src build -o -f \"(.ts|.tsx|.md|.html)$\"",
-    "build-components": "npm-run-all build:mitosis -p build-components:*",
-    "build-components:docs": "react-docgen ../../output/react/src/components/**/*.tsx -o ../../output/docs.json -i *.spec.tsx",
+    "build-components": "npm-run-all build:mitosis build-components:post build-components:docs",
```
Comment (mfranzke) [inline packages/components/package.json:20]: do all of those need to run sequentially, or could some run in parallel ?

File: output/stencil/scripts/packageLinkPhase.js [hunk 9]
```diff
@@ -46,11 +46,11 @@ const getFilteredContextData = (data) => {
 /**
  * Resolves props by data
  * @param resolvedData {object}
- * @param value {object}
+ * @param value {object|string}
  * @return {{resolvedType, name, type}}
  */
 const resolveProp = (resolvedData, value) => {
-	const { type } = value;
+	let type = value.type ? value.type : value;
```
Comment (mfranzke) [inline output/stencil/scripts/packageLinkPhase.js:53]:
```suggestion
	let type = value.type || value;
```

File: packages/components/src/shared/model.ts [hunk 10]
```diff
@@ -204,7 +207,7 @@ export const CustomValidityList = [
 export type CustomValidityType = (typeof CustomValidityList)[number];
 export type FormProps = {
 	/**
-	 * Marks an input element as invalid (red) | valid(green) | no-validation(grey). Overwrites the :user-valid selector.
+	 * Marks an input element as invalid (red) / valid(green) / no-validation(grey). Overwrites the :user-valid selector.
```
Comment (mfranzke) [inline packages/components/src/shared/model.ts:210]:
```suggestion
	 * Marks an input element as invalid (red) / valid (green) / no-validation (grey). Overwrites the :user-valid selector.
```

File: packages/components/src/shared/model.ts [hunk 11]
```diff
@@ -405,16 +408,49 @@ export const LinkReferrerPolicyList = [
...
+	/**
+	 * Sets aria attribute based on [`aria-current`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current).
+	 */
 	current?: boolean | LinkCurrentType;
+	/**
+	 * Disables the link.
+	 */
 	disabled?: boolean;
+	/**
+	 * The URL that the hyperlink points to. See [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href).
```
Comment (mfranzke) [inline packages/components/src/shared/model.ts:420]:
```suggestion
	 * The URL that the hyperlink points to. See [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href).
```

File: packages/components/src/shared/model.ts [hunk 12]
```diff
@@ -405,16 +408,49 @@ export const LinkReferrerPolicyList = [
...
+	 */
 	href?: string;
+	/**
+	 * Hints at the human language of the linked URL. See [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#hreflang).
```
Comment (mfranzke) [inline packages/components/src/shared/model.ts:424]:
```suggestion
	 * Hints at the human [language of the linked URL](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#hreflang).
```

File: packages/components/src/shared/model.ts [hunk 13]
```diff
@@ -405,16 +408,49 @@ export const LinkReferrerPolicyList = [
...
+	 */
 	hreflang?: string;
+	/**
+	 * Sets aria attribute based on [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label).
+	 */
 	label?: string;
+	/**
+	 * Where to display the linked URL, as the name for a browsing context. See [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target).
```
Comment (mfranzke) [inline packages/components/src/shared/model.ts:432]:
```suggestion
	 * Where to open the linked URL, as the name for a [browsing context](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target).
```

File: showcases/patternhub/components/old-routing-fallback/index.tsx [hunk 14]
```diff
@@ -32,7 +32,8 @@ const OldRoutingFallback = () => {
 				}
 
 				// This is for the old implementation to work with iframes
-				const foundRoute = getAllNavigationItems().find((item) =>
+				const allNavigationItems = getAllNavigationItems();
+				const foundRoute = allNavigationItems.find((item) =>
```
Comment (mfranzke) [inline showcases/patternhub/components/old-routing-fallback/index.tsx:36]: what's the reason for this change ?

Comment (nmerget) [reply showcases/patternhub/components/old-routing-fallback/index.tsx:36]: just for debugging, to set a breakpoint on line 36

File: packages/components/package.json (hunk 8 above)
Comment (nmerget) [reply packages/components/package.json:20]: Yes the docs may be affected by the post-build script. The post-build takes only some seconds so this shouldn't be a gib performance improvement

File: __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png
Comment (mfranzke) [inline __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png:1]: why is describedbyid removed? Did it become a global attribute ?

File: output/stencil/scripts/packageLinkPhase.js (hunk 9 above)
Comment (nmerget) [reply output/stencil/scripts/packageLinkPhase.js]: fixed

File: __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png
Comment (nmerget) [reply __snapshots__/brand/patternhub/brand-properties-should-match-screenshot.png:1]: I generate the `properties.mdx` out from `web-types.json` instead of `react-docgen`. 

The web-component anaylzer uses only props which are included inside the component. While some props e.g. `describedbyid` are inside the `GlobalProps`, but not used in the component `DBBrand`.

**Review by mfranzke — APPROVED** [review]

**Review by mfranzke — APPROVED** [review]
